File: frontend/demo/component/button/button-disabled.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/button';
import '@vaadin/horizontal-layout';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('button-disabled')
export class Example extends LitElement {
  protected createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  render() {
    return html`
      <vaadin-horizontal-layout theme="spacing">
        <!-- tag::snippet[] -->
        <vaadin-button theme="primary" disabled>Primary</vaadin-button>
        <vaadin-button theme="secondary" disabled>Secondary</vaadin-button>
        <vaadin-button theme="tertiary" disabled>Tertiary</vaadin-button>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/button/ButtonDisabled.java
package com.vaadin.demo.component.button;

import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.button.ButtonVariant;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("button-disabled")
public class ButtonDisabled extends Div {
    public ButtonDisabled() {
        // tag::snippet[]
        Button primaryButton = new Button("Primary");
        primaryButton.setEnabled(false);
        primaryButton.addThemeVariants(ButtonVariant.LUMO_PRIMARY);

        Button secondaryButton = new Button("Secondary");
        secondaryButton.setEnabled(false);

        Button tertiaryButton = new Button("Tertiary");
        tertiaryButton.setEnabled(false);
        tertiaryButton.addThemeVariants(ButtonVariant.LUMO_TERTIARY);
        // end::snippet[]

        HorizontalLayout horizontalLayout = new HorizontalLayout(primaryButton, secondaryButton, tertiaryButton);
        add(horizontalLayout);
    }

    public static class Exporter extends DemoExporter<ButtonDisabled> { // hidden-source-line
    } // hidden-source-line
}